components/onfleet/app/onfleet.app.ts

```diff
@@ -1,13 +1,225 @@
+import Onfleet from "@onfleet/node-onfleet";
 import { defineApp } from "@pipedream/types";
 
 export default defineApp({
   type: "app",
   app: "onfleet",
-  propDefinitions: {},
+  propDefinitions: {
+    apartment: {
+      type: "string",
+      label: "Address Line 2",
+      description: "For display purposes only. Used for Apartment, suite or other descriptive information.",
+      optional: true,
+    },
+    appearance: {
+      type: "object",
+      label: "Appearance",
+      description: "Set triangle colored pins on the map.",
+    },
+    autoAssign: {
+      type: "string",
+      label: "Assignment",
+      description: "Assignment type",
+      options: [
+        "auto-assign",
+        "team",
+        "driver",
+      ],
+    },
+    city: {
+      type: "string",
+      label: "Address City",
+      description: "If coordinates used, for display purposes only. Coordinates will be relied on for actual location.",
+    },
+    completeAfter: {
+      type: "integer",
+      label: "Complete After",
+      description: "A timestamp in unix time for the earliest time the task should be completed, in milliseconds precision.",
+    },
+    completeBefore: {
+      type: "integer",
+      label: "Complete Before",
+      description: "A timestamp in unix time for the latest time the task should be completed, in milliseconds precision.",
+    },
+    country: {
+      type: "string",
+      label: "Address Country",
+      description: "If coordinates used, for display purposes only. Coordinates will be relied on for actual location.",
+    },
+    dependencies: {
+      type: "string[]",
+      label: "Dependencies",
+      description: "An array of tasks which must be completed prior to this task.",
+      async options() {
+        const { tasks } = await this.listTasks({
+          from: 0,
+        });
+
+        return tasks.map(({ id }) => id);
```

Comment:
Wouldn't it be better to label the tasks with their names or some similar information, using the ids as the value?

Just a suggestion, not sure if this makes sense for this app